src/app_controller.js

```diff
@@ -713,23 +715,75 @@ AppController.prototype.init = function() {
        media: 'media/'});
 
   // Add tab handlers for switching between Block Factory and Block Exporter.
-  this.addTabHandlers(this.tabMap);
+  self.addTabHandlers(self.tabMap);
 
   // Assign exporter change listeners.
-  this.assignExporterChangeListeners();
+  self.assignExporterChangeListeners();
 
   // Create the root block on Block Factory main workspace.
   if ('BlocklyStorage' in window && window.location.hash.length > 1) {
     BlocklyStorage.retrieveXml(window.location.hash.substring(1),
                                BlockFactory.mainWorkspace);
   } else {
-    BlockFactory.showStarterBlock();
+    self.createBlocklyInitPopup(true);
   }
   BlockFactory.mainWorkspace.clearUndo();
 
   // Add Block Factory event listeners.
-  this.addBlockFactoryEventListeners();
+  self.addBlockFactoryEventListeners();
 
   // Workspace Factory init.
-  WorkspaceFactoryInit.initWorkspaceFactory(this.workspaceFactoryController);
+  WorkspaceFactoryInit.initWorkspaceFactory(self.workspaceFactoryController);
 };
+
+/**
+ * Creates popup for initializing blockly workspace, and then renders
+ * starter block.
+ *
+ * Helper function of init() and listener for Create New Block click.
+ * @param {boolean} whether function is being called on page load.
+ */
+AppController.prototype.createBlocklyInitPopup = function(firstLoad) {
+  var self = this;
+
+  $('#popup').css('display','inline');
+
+  if(!firstLoad) {
+    // Show exit button.
+    $('#exit').css('display','inline');
+
+    // Listener to x out popup.
+    $('#exit').click(function(){
+      $('#popup').css('display','none');
+    });
+  }
+
+  // Checks for block name type duplicates.
+  $('#block_name').change(function(){
+    if(self.blockLibraryController.has($("#block_name").val())) {
+      $("#block_name").css('border','1px solid red');
```

Comment:
Currently working on creating a FileManagerView and FileManagerController class for all block/project management (including popups for new blocks).